From ce23779ac5aeec75df91a4a2d176cf70bad21a55 Mon Sep 17 00:00:00 2001
From: papacu <papacu@papacu>
Date: Sat, 14 Jun 2025 02:04:10 +0000
Subject: [PATCH] remove markup

---
 Android.bp | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/Android.bp b/Android.bp
index 33fdd9c..d025c63 100644
--- a/Android.bp
+++ b/Android.bp
@@ -21,18 +21,6 @@ soong_namespace {
 	system_ext_specific: true,
 }
 
-android_app_import {
-	name: "MarkupGoogle_v2",
-	owner: "google",
-	apk: "proprietary/product/app/MarkupGoogle_v2/MarkupGoogle_v2.apk",
-	preprocessed: true,
-	presigned: true,
-	dex_preopt: {
-		enabled: false,
-	},
-	product_specific: true,
-}
-
 android_app_import {
 	name: "SoundPickerPrebuilt",
 	owner: "google",
-- 
2.43.0

