From b74d0156cf9957e2a9f8e035b67c60f26180d1bb Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Thu, 1 May 2025 15:36:13 +0100
Subject: [PATCH 1/1] Revert "gd: hci: Allow disabling selected local commands"

---
 system/gd/hci/controller.cc | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/system/gd/hci/controller.cc b/system/gd/hci/controller.cc
index b80cc65..b1d3d5c 100644
--- a/system/gd/hci/controller.cc
+++ b/system/gd/hci/controller.cc
@@ -16,7 +16,6 @@
 
 #include "hci/controller.h"
 
-#include <android-base/strings.h>
 #include <android_bluetooth_sysprop.h>
 #include <bluetooth/log.h>
 #include <com_android_bluetooth_flags.h>
@@ -49,8 +48,6 @@ constexpr bool kDefaultRpaOffload = false;
 static const std::string kPropertyVendorCapabilitiesEnabled =
         "bluetooth.core.le.vendor_capabilities.enabled";
 static const std::string kPropertyRpaOffload = "bluetooth.core.le.rpa_offload";
-static const char kPropertyDisabledCommands[] =
-        "bluetooth.hci.disabled_commands";
 
 constexpr bool kDefaultErroneousDataReportingEnabled = true;
 static const std::string kPropertyErroneousDataReportingEnabled =
@@ -370,15 +367,6 @@ struct Controller::impl {
     ErrorCode status = complete_view.GetStatus();
     log::assert_that(status == ErrorCode::SUCCESS, "Status {}", ErrorCodeText(status));
     local_supported_commands_ = complete_view.GetSupportedCommands();
-
-    if (auto disabledCommands = os::GetSystemProperty(kPropertyDisabledCommands)) {
-      for (const auto& command : android::base::Split(*disabledCommands, ",")) {
-        uint16_t index = std::stoi(command);
-        uint16_t byte_index = index / 10;
-        uint16_t bit_index = index % 10;
-        local_supported_commands_[byte_index] &= ~(1 << bit_index);
-      }
-    }
   }
 
   void read_local_extended_features_complete_handler(std::promise<void> promise,
-- 
2.47.2
