From 151a477802461685aacf56649e8fcc7f3979e7d9 Mon Sep 17 00:00:00 2001
From: papacu <papacu@apapacu.papacu>
Date: Mon, 16 Jun 2025 07:37:38 +0000
Subject: [PATCH 2/2] Fix bluetooth erro during compilation

---
 .../bluetooth/btservice/BondStateMachine.java        | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/android/app/src/com/android/bluetooth/btservice/BondStateMachine.java b/android/app/src/com/android/bluetooth/btservice/BondStateMachine.java
index 00c451190e..61a6670cd1 100644
--- a/android/app/src/com/android/bluetooth/btservice/BondStateMachine.java
+++ b/android/app/src/com/android/bluetooth/btservice/BondStateMachine.java
@@ -494,11 +494,11 @@ final class BondStateMachine extends StateMachine {
         return false;
     }
 
-    // Defining these properly would break current api
-    private static int PERIPHERAL_GAMEPAD = BluetoothClass.Device.Major.PERIPHERAL | 0x08;
-    private static int PERIPHERAL_REMOTE = BluetoothClass.Device.Major.PERIPHERAL | 0x0C;
+        // Defining these properly would break current api (não é verdade aqui — final é seguro)
+    private static final int PERIPHERAL_GAMEPAD = BluetoothClass.Device.Major.PERIPHERAL | 0x08;
+    private static final int PERIPHERAL_REMOTE = BluetoothClass.Device.Major.PERIPHERAL | 0x0C;
 
-    private static List<Pair<String, Integer>> accConfirmSkip = new ArrayList<>();
+    private static final List<Pair<String, Integer>> accConfirmSkip = new ArrayList<>();
 
     static {
         // Jarvis, SHIELD Remote 2015
@@ -508,7 +508,7 @@ final class BondStateMachine extends StateMachine {
     };
 
     @RequiresPermission(BLUETOOTH_CONNECT)
-    private boolean isSkipConfirmationAccessory(BluetoothDevice device) {
+    private static boolean isSkipConfirmationAccessory(BluetoothDevice device) {
         for (Pair<String, Integer> entry : accConfirmSkip) {
             String name = device.getName();
             if (name == null) {
@@ -519,10 +519,10 @@ final class BondStateMachine extends StateMachine {
                 return true;
             }
         }
-
         return false;
     }
 
+
     @RequiresPermission(allOf = {BLUETOOTH_CONNECT, BLUETOOTH_PRIVILEGED})
     private void sendDisplayPinIntent(byte[] address, Optional<Integer> maybePin, int variant) {
         BluetoothDevice device = mRemoteDevices.getDevice(address);
-- 
2.43.0

