From be6a8073be24b5618f576709d237b349d124f2ee Mon Sep 17 00:00:00 2001
From: papacu <papacu@papacu>
Date: Sat, 14 Jun 2025 01:56:36 +0000
Subject: [PATCH 5/5] remove certification

---
 config/derpfest.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/derpfest.mk b/config/derpfest.mk
index 3d1a355d..985f19b4 100644
--- a/config/derpfest.mk
+++ b/config/derpfest.mk
@@ -8,7 +8,7 @@ include vendor/fontage/config.mk
 include vendor/overlay/overlays.mk
 
 # Certification
-$(call inherit-product, vendor/certification/config.mk)
+# $(call inherit-product, vendor/certification/config.mk)
 
 # Additional props
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-- 
2.43.0

