From e8e515e9818a824b0e110ea3c0e4570abed4483b Mon Sep 17 00:00:00 2001
From: papacu <papacu@apapacu.papacu>
Date: Mon, 16 Jun 2025 05:46:44 +0000
Subject: [PATCH 7/7] remove more su and adb from evox

---
 config/common.mk | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/config/common.mk b/config/common.mk
index c6115180..1ea6666c 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -45,7 +45,7 @@ ifdef WITH_ADB_INSECURE
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += ro.adb.secure=0
 else
 # Enable ADB authentication
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += ro.adb.secure=1
+
 endif
 
 # Disable extra StrictMode features on all non-engineering builds
@@ -57,8 +57,6 @@ PRODUCT_COPY_FILES += \
     vendor/lineage/prebuilt/common/bin/backuptool.sh:install/bin/backuptool.sh \
     vendor/lineage/prebuilt/common/bin/backuptool.functions:install/bin/backuptool.functions
 
-PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
-    system/addon.d/50-lineage.sh
 
 ifneq ($(strip $(AB_OTA_PARTITIONS) $(AB_OTA_POSTINSTALL_CONFIG)),)
 PRODUCT_COPY_FILES += \
@@ -270,13 +268,6 @@ endif
 PRODUCT_PACKAGES += \
     adb_root
 ifneq ($(TARGET_BUILD_VARIANT),user)
-ifeq ($(WITH_SU),true)
-PRODUCT_PACKAGES += \
-    su
-
-PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
-    system/xbin/su
-endif
 endif
 
 # SystemUI
-- 
2.43.0

