From 1f5421f1a7d4a3d6ffce28876b326a1a7a374339 Mon Sep 17 00:00:00 2001
From: papacu <papacu@apapacu.papacu>
Date: Mon, 16 Jun 2025 04:11:29 +0000
Subject: [PATCH] remove cclibre

---
 datatop/src/Android.bp | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/datatop/src/Android.bp b/datatop/src/Android.bp
index 636f43d..0cd8fdd 100644
--- a/datatop/src/Android.bp
+++ b/datatop/src/Android.bp
@@ -2,16 +2,7 @@ cc_binary {
     name: "datatop",
     vendor: true,
 
-    cflags: [
-        "-Wall",
-        "-Wextra",
-        "-Werror",
-        "-pedantic",
-        "-std=c99",
-        "-DVERSION=\"1.0.4\"",
-        "-DHAVE_STRL_FUNCTIONS",
-        "-D_BSD_SOURCE",
-    ],
+
 
     srcs: [
         "datatop.c",
-- 
2.43.0

