From ac464269bb5e7d0dba9e68045858034c9d324fd0 Mon Sep 17 00:00:00 2001
From: Papacu <viniciussadxxx@gmail.com>
Date: Wed, 16 Jul 2025 15:53:25 -0300
Subject: [PATCH 17/17] fingerprint fix 2

---
 .../server/biometrics/AuthService.java        | 28 +++++++++----------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/services/core/java/com/android/server/biometrics/AuthService.java b/services/core/java/com/android/server/biometrics/AuthService.java
index 5ae5e840c..1e66e4ab8 100644
--- a/services/core/java/com/android/server/biometrics/AuthService.java
+++ b/services/core/java/com/android/server/biometrics/AuthService.java
@@ -917,20 +917,20 @@ public class AuthService extends SystemService {
             return;
         }
 
-+        final boolean preferHidlSensor;
-+        if (!SystemProperties.getBoolean("persist.sys.phh.virtual_sensors_are_real", false)) {
-+            if (hidlConfigStrings != null && hidlConfigStrings.length > 0
-+                    && fingerprintAidlInstances != null && fingerprintAidlInstances.length == 1
-+                    && fingerprintAidlInstances[0].contains("virtual")) {
-+
-+                Slog.w(TAG, "Preferring HIDL fingerprint sensor over virtual AIDL sensor");
-+                preferHidlSensor = true;
-+            } else {
-+                preferHidlSensor = false;
-+            }
-+        } else {
-+            preferHidlSensor = false;
-+        }
+        final boolean preferHidlSensor;
+        if (!SystemProperties.getBoolean("persist.sys.phh.virtual_sensors_are_real", false)) {
+            if (hidlConfigStrings != null && hidlConfigStrings.length > 0
+                    && fingerprintAidlInstances != null && fingerprintAidlInstances.length == 1
+                    && fingerprintAidlInstances[0].contains("virtual")) {
+
+                Slog.w(TAG, "Preferring HIDL fingerprint sensor over virtual AIDL sensor");
+                preferHidlSensor = true;
+            } else {
+                preferHidlSensor = false;
+            }
+        } else {
+            preferHidlSensor = false;
+        }
 
         handlerProvider.getFingerprintHandler().post(() -> {
             final FingerprintSensorConfigurations mFingerprintSensorConfigurations =
-- 
2.50.1

